boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -272,3 +279,7 @@ opm-build-push: docker-push
 	OPERATOR_IMAGE_TAG="${OPERATOR_IMAGE_TAG}" \
 	OLM_CHANNEL="${OLM_CHANNEL}" \
 	${CONVENTION_DIR}/build-opm-catalog.sh
+
+.PHONY: enable-fips
+enable-fips:
```

Comment:
Is the plan to have the consuming repo's Makefile invoke this somehow? Thinking about it, I wonder if it makes more sense to set it up so the consuming repo can just set a variable -- e.g. `FIPS_ENABLED:=true` -- and the rest gets done under the covers automatically. E.g. I would envision this target (which I would rename `ensure-fips`, and make idempotent as commented above) being included as a dep automatically in the appropriate places, something like:

```makefile
ifeq (${FIPS_ENABLED}, true)
go-generate: ensure-fips
endif
```

...and similar for other existing targets. (Note that dep lines like this are additive. I.e.
```makefile
foo: bar
foo: baz
```
is equivalent to
```makefile
foo: bar baz
```
so the above works to conditionally add the dep to the already-defined target after the fact.)

Reply:
The plan is to have each osd operator to have the option to opt into FIPS.
So once the operator updates to this version of boileplate, if they need FIPS they can run 'make enable-fips`  and they get these standardised files/flags etc.

FIPS will only be used in the dockerfile build and not locally

Reply:
That's pretty cool!

In my scenario here I would be adding a dep to go-build:
```.mk
ifeq (${FIPS_ENABLED}, true)
go-build: enable-fips
endif
```

The problem with having that variable set in the Makefile though is that on first run of `make go-build` locally the build will fail unless the user has the special go-toolset from the boilerplate image set as the GOROOT .
```.bash
$ make go-build
boilerplate/openshift/golang-osd-operator/configure-fips.sh
Writing fips file at /home/go/src/github.com/openshift/managed-upgrade-operator/cmd/manager/fips.go
# Force GOOS=linux as we may want to build containers in other *nix-like systems (ie darwin).
# This is temporary until a better container build method is developed
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 GOFLAGS="-tags=fips_enabled" GOOS=linux go build -gcflags="all=-trimpath=/home/johnroche/go" -asmflags="all=-trimpath=/home/johnroche/go" -o build/_output/bin/managed-upgrade-operator ./cmd/manager
cmd/manager/fips.go:9:2: package crypto/tls/fipsonly is not in GOROOT (/home/sdk/go1.16.7/src/crypto/tls/fipsonly)
```

I was trying to avoid having the need for that go-toolset to build locally.

The current way the consuming repo just:
- Updates boilerplate (+ commit)
- run `make enable-fips` (+ commit)

// Local builds use non fips go-toolset
// Dockerbuilds use fips

Does that make sense? That's what I was thinking about with that, open to suggestions on smarter ways to do that.

Reply:
You have described a canonical use case for [container-make](https://github.com/openshift/boilerplate/tree/master/boilerplate/openshift/golang-osd-operator#local-testing). It was designed exactly so that "local" builds could look just like CI and appsre builds, avoiding most of the problems that can arise from running on different platforms, with different (versions of) supporting tools installed, etc.

Put another way: you really shouldn't be counting on `make $target` for dev/test at all, not just because of FIPS.

Other than needing to install extra tools, is there some other reason local builds can't/shouldn't enable FIPS? E.g. does it significantly slow down the build? Does it refuse to build due to package entitlements?

Reply:
Let me double check with some more FIPS knowledgeable people. (MUO on ARO already uses FIPS) so
@hawkowl and @cblecker too

Any issues with/can we have the special go-toolset locally for dev?
Does running FIPS locally cause any issues?

Reply:
@2uasimojo I do agree with your approach though so i'll update with your suggestion.
qq on this, testing with the Makefile var, I had to put the var before the boilerplate includes.
e.g. In the consuming repo Makefile:
```.mk
FIPS_ENABLED=true

include boilerplate/generated-includes.mk
```
Is that acceptable or is there a way to do this I am unaware of?

Reply:
Fine by me 😁

Reply:
Perfect. And have updated with the suggested changes.
Let me know if you want the commit squashed.